src/lib/openjp2/ht_dec.c

```diff
@@ -1063,7 +1063,7 @@ static OPJ_BOOL opj_t1_allocate_buffers(
         if (flagssize > t1->flagssize) {
 
             opj_aligned_free(t1->flags);
-            t1->flags = (opj_flag_t*) opj_aligned_malloc(flagssize);
+            t1->flags = (opj_flag_t*) opj_aligned_malloc(flagssize * sizeof(opj_flag_t));
```

Comment:
at line 1074, the memset should also be changed to

memset(t1->flags, 0, flagssize * sizeof(opj_flag_t));

Reply:
You are right, I forget it.
I add this in new commit.